### PR TITLE
Feature emergency stop to dev

### PR DESCRIPTION
**Pull request by SamZand**
This is the emergency stop feature. We added two buttons to start en stop the movement. We refactored the typ manipulation by making a separate class for it. We created a queue where we put the commands. The module works now concurrent due to the loop function. There are test written and all the tests passes.

### CODE REVIEW

File: src/type_manipulation.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    while (*src) {
+        *dest++ = *src++;
+    }
+
+    *dest++ = '\0';
+
+    return saved;
+}
+
+char *TypeManipulation::intToChar(int number, char *dest) {
```
Comment (vasilishomegas) [inline src/type_manipulation.cpp]: Why return a char? It's a pointer that is passed as a parameter, so the recipient already knows it.

File: test/test_coordinate3d.hpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    REQUIRE(newCoord.y == 20);
+    REQUIRE(newCoord.z == -10);
+}
+
+TEST_CASE("Compare two Coordinate3D objects") {
+    RobotArm::Coordinate3D coord(10, 20, 30);
+    RobotArm::Coordinate3D coord2(10, 20, 30);
+
+    //REQUIRE(coord == RobotArm::Coordinate3D(10, 20, 30));
+    REQUIRE_FALSE((coord != coord2));
```
Comment (vasilishomegas) [inline test/test_coordinate3d.hpp]: Why the double negation?

File: src/type_manipulation.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    for (j = 0; src[j] != '\0'; j++) {
+        dest[i + j] = src[j];
+    }
+
+    dest[i + j] = '\0';
+    return dest;
+}
+
+char *TypeManipulation::strcopy(char *dest, const char *src) {
```
Comment (vasilishomegas) [inline src/type_manipulation.cpp]: Why return a char? It's a pointer that is passed as a parameter, so the recipient already knows it.

File: src/type_manipulation.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
+#include "type_manipulation.hpp"
+
+namespace RobotArm {
+char *TypeManipulation::stradd(char *dest, const char *src) {
```
Comment (vasilishomegas) [inline src/type_manipulation.cpp]: Why return a char? It's a pointer that is passed as a parameter, so the recipient already knows it.

Comment (vasilishomegas) [reply src/type_manipulation.cpp]: I also find it more logical to first mention the source and then the destination.

File: test/test_main.cpp [hunk 5]
```diff
@@ -1,73 +1,32 @@
...
  * @file
  * @brief     Test file
- * @author    Jeroen van Hattem
+ * @author    Jeroen van Hattem, Wiebe van Breukelen
  * @license   MIT
  */
 #define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
-#include "../src/coordinate3d.hpp"
+
 #include "../src/robot_arm.hpp"
```
Comment (vasilishomegas) [inline test/test_main.cpp]: Can't you include these directly? So without ```../src/```?

**Review by vasilishomegas — CHANGES_REQUESTED** [review]

File: src/type_manipulation.cpp (hunk 1 above)
Comment (SamZand) [reply src/type_manipulation.cpp]: Fair point.
done

File: test/test_coordinate3d.hpp (hunk 2 above)
Comment (SamZand) [reply test/test_coordinate3d.hpp]: Fair point.

File: src/type_manipulation.cpp (hunk 3 above)
Comment (SamZand) [reply src/type_manipulation.cpp]: Again Fair point.
Done

File: src/type_manipulation.cpp (hunk 4 above)
Comment (SamZand) [reply src/type_manipulation.cpp]: ok Fair point about the return a char.
I agree with the fact that it looks beter to swap dest en src but the amount of work vs what we gain is in our opinion to little.

File: test/test_main.cpp (hunk 5 above)
Comment (SamZand) [reply test/test_main.cpp]: Done!

File: test/test_coordinate3d.hpp (hunk 2 above)
Comment (SamZand) [reply test/test_coordinate3d.hpp]: The test for a != operator which requires false. So thats why.. :)

**Review by vasilishomegas — APPROVED** [review]

**Pull request merged**